src/DomainManager.Bussines/Notifications/CommandConsumers/PortScannerCommandConsumer.cs

```diff
@@ -0,0 +1,43 @@
+using System.Net.Sockets;
+using DomainManager.Abstract;
+using DomainManager.Notifications.CommandConsumers.Base;
+using MassTransit.Mediator;
+using Microsoft.Extensions.Caching.Memory;
+using Telegram.Bot;
+
+namespace DomainManager.Notifications.CommandConsumers;
+
+public class PortScanner : CommandConsumerBase, IMediatorConsumer {
+    private readonly ApplicationDbContext _db;
+    private readonly IScopedMediator _mediator;
+
+    public PortScanner(ITelegramBotClient botClient, ApplicationDbContext db, IScopedMediator mediator,
+        IMemoryCache memoryCache) :
+        base(Command.PortScan, botClient, memoryCache) {
+        _db = db;
+        _mediator = mediator;
+    }
+
+    protected override async Task<string> Consume(string[] args, Message message, long chatId, bool isAdmin,
+        CancellationToken cancellationToken) {
+
+        return args switch {
+            [.., { } host, { } port] when isAdmin => await GetHostList(host, port, cancellationToken),
```

Comment:
Что за HostList?)
```suggestion
            [.., { } host, { } portStr] when isAdmin && int.TryParse(portStr, out var port)  => await ScanPort(host, port, cancellationToken),
```
так если порт будет не числом то получим help